test/support/utils.ts

```diff
@@ -59,3 +59,32 @@ export function promisifyChildProcess(
     })
   })
 }
+
+export function waitUntil(
+  expectation: () => void,
+  { timeout = 3000, interval = 1000 } = {},
+) {
+  if (interval < 1) interval = 1
+  const maxTries = Math.ceil(timeout / interval)
+  let tries = 0
+  return new Promise((resolve, reject) => {
+    const rejectOrRerun = (error: unknown) => {
+      if (tries > maxTries) {
+        reject(error)
+        return
+      }
+      setTimeout(runExpectation, interval)
+    }
+    function runExpectation() {
+      tries += 1
+      try {
+        Promise.resolve(expectation())
+          .then(() => resolve(undefined))
+          .catch(rejectOrRerun)
+      } catch (error: unknown) {
```

Comment:
Does TypeScript require we type the `error` here? Afaik, it should be inferred as a general `Error`. If it's not, we should annotate it as such:

```ts
} catch (error: Error) {
```

Reply:
no we don't need